Description: Fix thread priority test
 Test should not assume to be run at priority 0.
 .
 See upstream bug for more details.
Bug: https://github.com/libuv/libuv/issues/4382
Bug-Debian: https://bugs.debian.org/1071194
Author: info@bnoordhuis.nl
Origin: https://github.com/libuv/libuv/issues/4382
Applied-Upstream: not-needed
--- a/test/test-thread-priority.c
+++ b/test/test-thread-priority.c
@@ -89,10 +89,12 @@
 */
 #ifdef __linux__
   ASSERT_OK(uv_thread_getpriority(pthread_self(), &priority));
-  ASSERT_EQ(priority, 0);
-  ASSERT_OK(uv_thread_setpriority(pthread_self(), UV_THREAD_PRIORITY_LOWEST));
-  ASSERT_OK(uv_thread_getpriority(pthread_self(), &priority));
-  ASSERT_EQ(priority, (0 - UV_THREAD_PRIORITY_LOWEST * 2));
+  if (priority == 0 ) {
+      // do not run test when process is run with nice (seen on Debian)
+      ASSERT_OK(uv_thread_setpriority(pthread_self(), UV_THREAD_PRIORITY_LOWEST));
+      ASSERT_OK(uv_thread_getpriority(pthread_self(), &priority));
+      ASSERT_EQ(priority, (0 - UV_THREAD_PRIORITY_LOWEST * 2));
+  }
 #endif
 
   uv_sem_post(&sem);
@@ -102,4 +104,4 @@
   uv_sem_destroy(&sem);
 
   return 0;
-}
\ No newline at end of file
+}
